src/analysis/normed_space/deriv.lean

```diff
@@ -0,0 +1,178 @@
+/-
+Copyright (c) 2019 Jeremy Avigad. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Jeremy Avigad
+
+The Fréchet derivative.
+
+Let `E` and `F` be normed spaces, and `f : E → F`. Then
+
+  `has_fderiv_at_within f f' x s`
+
+says that the function `f' : E → F` is a bounded linear map and `f` has derivative `f'` at
+`x`, where the domain of interest is restricted to `s`. We also have
+
+  `has_fderiv_at f f' x := has_fderiv_at_within f f' x univ`
+
+We may want to prove some theorems specifically for this special case, but most of the more
+general theorems should apply right out of the box.
+
+The derivative is characterized in two ways: in terms of the `tendsto` relation, and in terms of
+the `littleo` relation.
+-/
+import topology.basic analysis.normed_space.bounded_linear_maps ..asymptotics tactic.abel
+open filter asymptotics
+
+variables {E : Type*} [normed_space ℝ E]
+variables {F : Type*} [normed_space ℝ F]
+variables {G : Type*} [normed_space ℝ G]
+
+def has_fderiv_at_within (f : E → F) (f' : E → F) (x : E) (s : set E) :=
```

Comment:
Why is this called `has_fderiv_at_within` and not `is_fderiv_at_withing`?

Reply:
The identifier and arguments should reflect what you would say verbally: "f has (Fréchet) derivative f' at x within s."